File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Effect/Combat/CleaveEffect.ts
import { FairyGUI } from "csharp";
import { Mathf } from "../../../../../Core/Module/Math/Mathf";
import { TimeKit } from "../../../../../Core/Utils/TimeKit";
import { View_CleaveEffect } from "../../Gen/View/ModTheSpire_Effect";
import { AbstractEffect } from "../AbstractEffect";


export class CleaveEffect extends AbstractEffect{
    private Loader:FairyGUI.GComponent;
    private StallTimer:number;
    private Scale:number;
    public constructor(private _X : number,private _Y:number) {
        super();
        this.Loader.SetPosition(_X,_Y,0);
        this.Loader = <FairyGUI.GComponent>View_CleaveEffect.CreateInstance();
        this.Loader.rotation = Mathf.RandomRange(-5,1);
        this.Scale = 1.2;
        this.Loader.SetScale(this.Scale,this.Scale)
        this.StallTimer = Mathf.RandomRange(0,0.2);
    }
    Update() {
        super.Update();
        if(this.StallTimer > 0){
            this.StallTimer -= TimeKit.DeltaTime;
            return;
        }
        this.Scale+= 0.005;
        this.Loader.SetScale(this.Scale,this.Scale);
        this.Loader.rotation += Mathf.RandomRange(-0.5,0.5);
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Effect/Combat/CleaveEffect.cs
using FairyGUI;
using ModTheSpire.ModTheSpire_Effect;
using UnityEngine;

namespace ModTheSpire.Effect.Combat
{
    public class CleaveEffect : AbstractEffect
    {
        private GComponent mLoader;
        private float mStallTimer;
        private float mScale;
        public CleaveEffect(float x,float y)
        {
            mLoader = View_CleaveEffect.CreateInstance();
            mLoader.SetPosition(x,y,0);

            mLoader.rotation = Random.Range(-5,1);
            mScale = 1.2f;
            mLoader.SetScale(mScale, mScale);
            mStallTimer = Random.Range(0,0.2f);
        }

        public override void Update()
        {
            base.Update();
            if(mStallTimer > 0){
                mStallTimer -= Time.deltaTime;
                return;
            }

            mScale += 0.005f;
            mLoader.SetScale(mScale,mScale);
            mLoader.rotation += Random.Range(-0.5f, 0.5f);
        }
    }
}